### ISSUE TITLE
Issue with rigged model

### ISSUE THREAD

**Issue by fullmoon8** [niacdoial/blemd#36]
Hey
Recently I've been getting into modding SMS, and decided to try replacing mario's model with Sans for fun.

I used BleMD to import mario's model and armature to Blender, then deleted the model and weighted a model of sans to the bones. Testing it with mario's idle animation, it looks like this: https://i.gyazo.com/89d27c6a4ea5230ce98c6156b4b6ccc3.mp4

However, after putting the model into the game, the model appears like this: https://i.gyazo.com/4c027dfb221e0a8e05d31035040f79b8.mp4

I've been trying to figure out a workaround to this for a few days to no avail :/
Anyone know what's going wrong here?

**Comment by niacdoial** [comment]
OK.
first off,
> decided to try replacing mario's model with Sans for fun
I see you're a [person] of culture as well.

I think that's due to the (in retrospect overcomplicated) way blemd changes the Y-up BMD coordinates into the Z-up blender coordinates.
I think you might want to use the "use natural bone placement" option on import.
It will discard animations, but it will be easier to re-export.
Just watch out for the armature's rotation.

Hope that helps.

Also, what tools did you use to put the model back into the game? it would help me test BleMD as a bmd exporter if/when I reach that point.

**Comment by fullmoon8** [comment]
> I see you're a [person] of culture as well.

I really don't get why it wasn't Super Sans Sunshine to begin with! Really sucks when us fans have to fix mistakes for big companies like nintendo. smh

I tried it with the natural bone placement and reweighted, but it imports with less bones than normal, and it ends up looking like this in-game lol: https://i.gyazo.com/d55e7d9e534230daa9339ce515c5a9f1.mp4

Do you think this could be fixed by rotating something?

Btw I export as DAE, and then use SuperBMD to convert the model to bmd.
I use rarcdump to unpack mario's .szs file, and then replace his model with the new one.
I use ArcPack and yaz0fast to get the whole file back to .szs format, and then rebuild the iso with GCRebuilder.

**Comment by niacdoial** [comment]
OK. So, first off, thanks. I'll be looking into those tools at some point or another.<br/>
By the way here is a version of BMDView2 (compiled for windows), <br/>
It's a pretty great bmd+animation viewer, so it will help look at SuperBMD's output more directly and faster. <br/>
I included the source in case you wanted to tinker with it or recompile it.

second....
My bad, I messed several things up. It's really been too long since I've worked on this.
The option I wanted you to use somehow never left my own PC...
It's in the newest commit now. It's called "disable axis conversion", not "use natural bone placement"

sorry!

Hope this is better
